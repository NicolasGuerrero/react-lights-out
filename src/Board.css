.Board {
  margin: 0 auto;
  display: inline-block;
  width: auto;
  text-align: center;
}