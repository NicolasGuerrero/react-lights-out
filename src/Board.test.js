import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Board from "./Board";

it("renders without crashing", function () {
  render(<Board />);
});

it("matches snapshot - new board", function () {
  const { asFragment } = render(<Board chanceLightStartsOn={1}/>);
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot - winning", function () {
  const { asFragment } = render(<Board chanceLightStartsOn={0}/>);
  expect(asFragment()).toMatchSnapshot();
});

it("creates starter board", function () {
  const { queryByTestId } = render(<Board nrows={3} ncols={3} chanceLightStartsOn={1} />);
  const middleCell = queryByTestId('1-1');
  expect(middleCell).toBeInTheDocument();

  fireEvent.click(middleCell);

  expect(queryByTestId('0-0')).toHaveClass('Cell-lit');
  expect(queryByTestId('0-1')).not.toHaveClass('Cell-lit');
  expect(queryByTestId('0-2')).toHaveClass('Cell-lit');
  expect(queryByTestId('1-0')).not.toHaveClass('Cell-lit');
  expect(queryByTestId('1-1')).not.toHaveClass('Cell-lit');
  expect(queryByTestId('1-2')).not.toHaveClass('Cell-lit');
  expect(queryByTestId('2-0')).toHaveClass('Cell-lit');
  expect(queryByTestId('2-1')).not.toHaveClass('Cell-lit');
  expect(queryByTestId('2-2')).toHaveClass('Cell-lit');
});

it("should flash you win message", function () {
  const { getByText } = render(<Board chanceLightStartsOn={0} />);
  
  expect(getByText('You Won!')).toBeInTheDocument();
});