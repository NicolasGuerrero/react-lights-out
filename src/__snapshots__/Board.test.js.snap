// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot - new board 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <td
          class="Cell Cell-lit"
          data-testid="0-0"
        />
        <td
          class="Cell Cell-lit"
          data-testid="0-1"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
          data-testid="1-0"
        />
        <td
          class="Cell Cell-lit"
          data-testid="1-1"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`matches snapshot - winning 1`] = `
<DocumentFragment>
  <p>
    You Won!
  </p>
</DocumentFragment>
`;
